File: coding_challenges/cd/cd.cpp
#include <iostream>
using namespace std;

int main() {
  int N, M;
  for(;;) {
    cin >> N >> M;
    if (N == 0 && M == 0) {
      break;
    }
    int A[N];
    int B[M];

    for (int i=0; i < N; i++) {
      cin >> A[i];
    }
    for (int i=0; i < M; i++) {
      cin >> B[i];
    }

    int i = 0, j = 0, result = 0;
    while(i < N && j < M) {
      // cout << "i:" << i << " j:" << j << endl;
      if (A[i] == B[j]) {
        result++; i++; j++;
      } else if (A[i] > B[j]) {
        j++;
      } else {
        i++;
      }
    }
    cout << result << endl;
  }
  return 0;
}

File: coding_challenges/cd/cd.py
def in_common_sets(A, B):
    print(len(A & B))

def in_common(A, B):
    i = 0
    j = 0
    result = 0
    while (i < len(A) and j < len(B)):
        # print(i, j)
        if (A[i] == B[j]):
            result += 1
            i += 1
            j += 1
        elif (A[i] > B[j]):
            j += 1
        else:
            i += 1
    print(result)

def main():
    filename = "cd_test_1.txt"
    file = open(filename)
    number = file.readline()
    number = number.split()
    set1 = set()
    set2 = set()
    list1 = []
    list2 = []
    for x in range(0, int(number[0])):
        line = file.readline()
        set1.add(line.strip())
        list1.append(line.strip())
    for x in range(0, int(number[1])):
        line = file.readline()
        set2.add(line.strip())
        list2.append(line.strip())
    # print(list1)
    # print(list2)
    in_common_sets(set1, set2)
    in_common(list1, list2)

main()